### ISSUE TITLE
Is this CBAM correct？

### ISSUE THREAD

**Issue by q1135718080** [AlexeyAB/darknet#8419]
I try to implement CBAM with Pytorch, but I don't know if it's written correctly. Can you help me see it
Thank you very much.
`# Channel attention
class CAP(nn.Module):
    def __init__(self, in_planes, ratio=16):
        super(CAP, self).__init__()
        self.avg_pool = nn.AdaptiveAvgPool2d(1)
        self.max_pool = nn.AdaptiveMaxPool2d(1)

        self.fc1 = nn.Conv2d(in_planes, in_planes // 16, 1, bias=False)
        self.relu1 = nn.ReLU()
        self.fc2 = nn.Conv2d(in_planes // 16, in_planes, 1, bias=False)

        self.sigmoid = nn.Sigmoid()

    def forward(self, x):
        avg_out = self.fc2(self.relu1(self.fc1(self.avg_pool(x))))
        max_out = self.fc2(self.relu1(self.fc1(self.max_pool(x))))
        out = avg_out + max_out
        return self.sigmoid(out)`

`        elif mdef['type'] == 'cap':
            modules = CAP()`

`######SAM######
[cap]

[convolutional]
batch_normalize=1
size=1
stride=1
pad=1
filters=512
activation=logistic

[sam]
from=-2
######SAM######`

**Comment by q1135718080** [comment]
```
#######CBAM#######
[avgpool]

[convolutional]
filters=16
size=1
stride=1
activation=linear

[convolutional]
filters=256
size=1
stride=1
activation=linear

[route]
layers=-4

[maxpool]

[convolutional]
filters=16
size=1
stride=1
activation=linear

[convolutional]
filters=256
size=1
stride=1
activation=linear

[route]
layers=-1,-5

[Sigmoid]

[convolutional]
batch_normalize=1
size=1
stride=1
pad=1
filters=256
activation=logistic

[sam]
from=-2
#######CBAM#######
```

**Comment by ryotaro137** [comment]
I don't know much about it, but please tell me why you are adding the SAM at the end.
Thank you in advance.
reference:https://github.com/luuuyi/CBAM.PyTorch
